xarray/backends/scipy_.py

```diff
@@ -96,28 +110,38 @@ def __init__(self, filename_or_obj, mode='r', format=None, group=None,
             raise ValueError('invalid format for scipy.io.netcdf backend: %r'
                              % format)
 
+        # if the string ends with .gz, then gunzip and open as netcdf file
+        if type(filename_or_obj) is str and filename_or_obj.endswith('.gz'):
```

Comment:
Use `isinstance(filename_or_obj, basestring)` instead of `type(filename_or_obj) is str`.

Also, move this logic into `_open_scipy_netcdf` -- otherwise it won't work to reopen a gzipped file.